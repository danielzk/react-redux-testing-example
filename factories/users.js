import {Factory} from 'rosie'
import {OrderedMap} from 'immutable'

export const UserFactory = new Factory()
  .sequence('id', sequence => sequence)
  .sequence('username', sequence => 'username' + sequence)
  .attr('firstName', 'Mark')
  .attr('lastName', 'Zucaritas')

export const UserListFactory = (count) => {
  const users = UserFactory.buildList(count)
  return OrderedMap(users.map(item => [item.id, item]))
}
