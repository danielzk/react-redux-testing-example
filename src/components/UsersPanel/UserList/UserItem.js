import React from 'react'

export default class UserItem extends React.Component {
  onDeleteClick = () => {
    this.props.actions.deleteUser(this.props.user.id)
  }

  render() {
    return (
      <li>
        <span>Username: {this.props.user.username}</span>
        <span>Full name: {this.props.user.firstName} {this.props.user.lastName}</span>
        <button onClick={this.onDeleteClick}>Delete</button>
      </li>
    )
  }
}
