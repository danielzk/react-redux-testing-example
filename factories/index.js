import {Factory} from 'rosie'

import {UserFactory, UserListFactory} from './users'

export const StoreFactory = new Factory()
  .option('numUsers', 2)
  .attr('users', ['numUsers'], count => UserListFactory(count))

export {UserFactory, UserListFactory}
