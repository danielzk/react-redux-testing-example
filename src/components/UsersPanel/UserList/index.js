import React from 'react'

import UserItem from './UserItem'

export default class UserList extends React.Component {
  render() {
    return (
      <div>
        {this.props.users.count() > 0 &&
          <ul>
            {this.props.users.toArray().map(user =>
              <UserItem user={user} key={user.id} {...this.props}/>
            )}
          </ul>
        }

        {this.props.users.count() == 0 &&
          <span>No users</span>
        }
      </div>
    )
  }
}
