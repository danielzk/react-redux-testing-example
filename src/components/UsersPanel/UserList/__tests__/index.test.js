import React from 'react'
import {OrderedMap} from 'immutable'
import {shallow} from 'enzyme'

import {UserListFactory} from '../../../../../factories'
import UserList from '../'

describe('UserList component', () => {
  it('should render list of users', () => {
    const props = {users: UserListFactory(2)}
    const wrapper = shallow(<UserList {...props}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should display a no-users message if no users', () => {
    const props = {users: OrderedMap()}
    const wrapper = shallow(<UserList {...props}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
