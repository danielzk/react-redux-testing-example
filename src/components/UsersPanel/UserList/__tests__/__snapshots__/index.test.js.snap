// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component should display a no-users message if no users 1`] = `
<div>
  <span>
    No users
  </span>
</div>
`;

exports[`UserList component should render list of users 1`] = `
<div>
  <ul>
    <UserItem
      key="1"
      user={
        Object {
          "firstName": "Mark",
          "id": 1,
          "lastName": "Zucaritas",
          "username": "username1",
        }
      }
      users={
        Immutable.OrderedMap {
          1: Object {
            "firstName": "Mark",
            "id": 1,
            "lastName": "Zucaritas",
            "username": "username1",
          },
          2: Object {
            "firstName": "Mark",
            "id": 2,
            "lastName": "Zucaritas",
            "username": "username2",
          },
        }
      }
    />
    <UserItem
      key="2"
      user={
        Object {
          "firstName": "Mark",
          "id": 2,
          "lastName": "Zucaritas",
          "username": "username2",
        }
      }
      users={
        Immutable.OrderedMap {
          1: Object {
            "firstName": "Mark",
            "id": 1,
            "lastName": "Zucaritas",
            "username": "username1",
          },
          2: Object {
            "firstName": "Mark",
            "id": 2,
            "lastName": "Zucaritas",
            "username": "username2",
          },
        }
      }
    />
  </ul>
</div>
`;
